packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -194,6 +197,58 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
         }
     }
 
+    // IDs of the most recently used terminals
+    protected mostRecentlyUsedTerminalEntries: { id: string, disposables: DisposableCollection }[] = [];
+
+    protected getLastUsedTerminalId(): string | undefined {
+        const mostRecent = this.mostRecentlyUsedTerminalEntries[this.mostRecentlyUsedTerminalEntries.length - 1];
+        if (mostRecent) {
+            return mostRecent.id;
+        }
+    }
+
+    get lastUsedTerminal(): TerminalWidget | undefined {
+        const id = this.getLastUsedTerminalId();
+        if (id) {
+            return this.getById(id);
+        }
+    }
+
+    protected setLastUsedTerminal(lastUsedTerminal: TerminalWidget): void {
+        const lastUsedTerminalId = lastUsedTerminal.id;
+        const entryIndex = this.mostRecentlyUsedTerminalEntries.findIndex(entry => entry.id === lastUsedTerminalId);
+        let toDispose: DisposableCollection | undefined;
+        if (entryIndex >= 0) {
+            toDispose = this.mostRecentlyUsedTerminalEntries[entryIndex].disposables;
+            this.mostRecentlyUsedTerminalEntries.splice(entryIndex, 1);
+        } else {
+            toDispose = new DisposableCollection();
+            toDispose.push(
```

Comment:
I would expect that we hook into ondidcreate hook of WidgetManager and install a listener where for each terminal always.

Reply:
Nvm I see that you do it already )